src/Store/reducer.js

```diff
@@ -0,0 +1,35 @@
+import HttpGateway from '../Checkbox/HttpGateway';
+
+function makeSetIsChecked(httpGateway) {
```

Comment:
This uses dependency injection to inject the `httpGateway` into the reducer.

The fact that it's not OOP doesn't matter. That's not needed for DI.